src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -372,6 +427,30 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
             except ApiError as e:
                 logger.exception("slack.action.response-error", extra={"error": str(e)})
 
+    def open_archive_dialog(self, slack_request: SlackActionRequest, group: Group) -> None:
+        callback_id = {
+            "issue": group.id,
+            "orig_response_url": slack_request.data["response_url"],
+            "is_message": _is_message(slack_request.data),
+        }
+
+        if slack_request.data.get("channel"):
+            callback_id["channel_id"] = slack_request.data["channel"]["id"]
+            callback_id["rule"] = slack_request.callback_data.get("rule")
```

Comment:
nit: I think this can be in the initial `callback_id` creation since it's not relying on having channel data